src/foundry/client/pixi/layers/grid/grid.d.ts

```diff
@@ -1,171 +1,249 @@
+     * @param y  - The starting y-coordinate in pixels
+     * @param dx - The number of grid positions to shift horizontally
+     * @param dy - The number of grid positions to shift vertically
+     * @param options - Additional options to configure shift behaviour.
      */
-    y?: number;
+    shiftPosition(
+      x: number,
+      y: number,
+      dx: number,
+      dy: number,
+      options?: {
+        /**
+         * The token that is being shifted.
+         */
+        token?: Token;
+      },
+    ): PointArray;
 
     /**
-     * The hex fill color of the highlight
-     * @defaultValue `0x33BBFF`
+     * Measure the distance traversed over an array of measured segments
+     * @param segments - An Array of measured movement segments
+     * @param options  - Additional options which modify the measurement
+     *                   (default: `{}`)
+     * @returns An Array of distance measurements for each segment
      */
-    color?: number;
+    measureDistances(segments: GridLayer.Segment[], options?: MeasureDistancesOptions | undefined): number[];
 
     /**
-     * The hex border color of the highlight
-     * @defaultValue `null`
+     * Get the grid row and column positions which are neighbors of a certain position
+     * @param row - The grid row coordinate against which to test for neighbors
+     * @param col - The grid column coordinate against which to test for neighbors
+     * @returns An array of grid positions which are neighbors of the row and column
      */
-    border?: number | null;
+    getNeighbors(row: number, col: number): PointArray[];
 
     /**
-     * The opacity of the highlight
-     * @defaultValue `0.25`
+     * Determine a placeable's bounding box based on the size of the grid.
+     * @param w - The width in grid spaces.
+     * @param h - The height in grid spaces.
      */
-    alpha?: number;
+    getRect(w: number, h: number): PIXI.Rectangle;
 
     /**
-     * A predefined shape to highlight
-     * @defaultValue `null`
+     * Calculate the resulting token position after moving along a ruler segment.
+     * @param ray    - The ray being moved along.
+     * @param offset - The offset of the ruler's origin relative to the token's position.
+     * @param token  - The token placeable being moved.
+     * @internal
      */
-    shape?: PIXI.Polygon | null;
+    protected _getRulerDestination(ray: Ray, offset: Point, token: Token): { x: number; y: number };
+  }
+
+  namespace BaseGrid {
+    interface GridOptions {
+      dimensions: Canvas["dimensions"];
+      color: string;
+      alpha: Scene["data"]["gridAlpha"];
+      columns?: boolean;
+      even?: boolean;
+    }
+
+    interface DrawOptions {
+      /**
+       * The grid color.
+       * @defaultValue `null`
+       */
+      color?: string | null | undefined;
+
+      /**
+       * The grid transparency.
+       * @defaultValue `null`
+       */
+      alpha?: number | null | undefined;
+    }
```

Comment:
These interfaces are perfectly fine because they already have `undefined` everywhere there's a `?`. But if you want you can use `InexactPartial` as `interface DrawOptions extends InexactPartial<{..}> {}` but that's not that clean either.